### PR TITLE
[feature] Automate generation of test-translation.json from .po files

### LINKED ISSUE #902: [feature] Automate generation of client/test-translation.json from .po files
**Is your feature request related to a problem? Please describe.**
The `client/test-translation.json` file currently requires manual updates to add or change translations for strings. This is repetitive work since the translations already exist in the corresponding `.po` files. If one misses updating test-translation.json file, then the snapshots created for tests only include the `msgid`. This creates inconsistency and increases maintenance overhead. 

**Describe the solution you'd like**
A script or tool that can automatically generate or update `client/test-translation.json` by extracting translations directly from the `.po` files.

 A possible solution could be to add this process to the `yarn test` command, so the file is populated automatically before tests are run.

### PR DESCRIPTION
**Pull request by Ajay110125yadav**
This PR implements automatic generation of test-translation.json from .po files as requested in #902.

Changes:
- Added `scripts/generate-test-translations.js` to parse .po files and extract translations
- Handles both single-line and multi-line msgstr formats
- Updated test-translation.json with complete translations from i18n/en.po
- Fixed test that was expecting untranslated key

Testing:
- All tests pass locally
- yarn test -u used to update snapshots

Closes #902

### CODE REVIEW

File: scripts/generate-test-translations.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    continue;
+  }
+
+  // Continue collecting multi-line msgstr
+  if (isCollectingMsgstr && line.trim().startsWith('"')) {
+    const content = line.trim().substring(1, line.trim().length - 1);
+    currentMsgstr += content;
+    continue;
+  }
```
[the remark this answers is not in the document]
Comment (Ajay110125yadav) [reply scripts/generate-test-translations.js]: Thanks for the detailed feedback! The current implementation
handles the main use cases for issue #902. I'll create a follow-up
issue to implement gettext-parser for better edge case handling
if this approach is accepted.
